lib/puppet/util/monkey_patches.rb

```diff
@@ -20,6 +20,9 @@ def self.caller(skip=nil)
 
 class Symbol
   def <=> (other)
+    if (other.class != Symbol)
```

Comment:
Yeah. I wanted to benchmark how bad it was to add a check for `Puppet[:strict]` here, and I was gonna check the base "no warnings at all" case, too